app/src/sharedTest/java/org/oppia/app/splash/SplashActivityTest.kt

```diff
@@ -0,0 +1,47 @@
+package org.oppia.app.splash
+
+import androidx.fragment.app.Fragment
+import androidx.test.espresso.intent.Intents.intended
+import androidx.test.espresso.intent.matcher.IntentMatchers.hasComponent
+import androidx.test.espresso.intent.rule.IntentsTestRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.HomeActivity
+
+/** Tests for [SplashActivity]. */
+@RunWith(AndroidJUnit4::class)
+class SplashActivityTest {
+
+  @get:Rule
+  var mActivityRule: IntentsTestRule<SplashActivity> = IntentsTestRule(
+    SplashActivity::class.java
+  )
+  var mainActivity: SplashActivity? = null
+  var mainActivityFragment: SplashFragment? = null
+
+  @Before
+  fun setUp() {
+    mainActivity = mActivityRule.getActivity();
+    mainActivityFragment = SplashFragment()
+    startFragment(mainActivityFragment!!)
+  }
+
+  @Test
+  fun testMainActivity() {
+    Assert.assertNotNull(mainActivity)
+  }
+
+  private fun startFragment(fragment: Fragment) {
```

Comment:
Why is this needed? If we're creating SplashActivity directly, shouldn't that already have splash fragment created & added?

Reply:
It throws an error 
`androidx.test.espresso.base.DefaultFailureHandler$AssertionFailedWithCauseError: Wanted to match 1 intents. Actually matched 0 intents.

IntentMatcher: has component: has component with: class name: is "org.oppia.app.HomeActivity" package name: an instance of java.lang.String short class name: an instance of java.lang.String`
if I don't start a fragment over here

Reply:
I have made use of ActivityTestRule<SplashActivity>. I was assuming that I need to launch fragment and then it will launch HomeActivity. 

It is passing testcase now. Code is updated